### ISSUE TITLE
 building an ig is giving me a heap space error.

### ISSUE THREAD

**Issue by Healthedata1** [qligier/fhir-ig-action#5]
Building an ig is giving me a heap space error.  In my local build I allocate 4G: 

`java -Xmx4G -jar ${path} -ig ig.ini -tx $NA`

Can you add a parameter to adjust the to increase the heap maximum size or hardcode it.  I am not sure if you need paid accounts or not to do this.

From Github
>[Supported runners and hardware resources](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)
Note: GitHub also offers larger runners, which are available in larger configurations, with autoscaling enabled by default and optional dedicated IP addresses. For more information, see "[About larger runners](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners#machine-specs-for-larger-runners)."

>Hardware specification for Windows and Linux virtual machines:
>2-core CPU (x86_64)
>7 GB of RAM
>14 GB of SSD space

>Hardware specification for macOS virtual machines:
>3-core CPU (x86_64)
>14 GB of RAM
>14 GB of SSD space

>Hardware specification for macOS XL virtual machines:
>12-core CPU (x86_64)
>30 GB of RAM
>14 GB of SSD space

**Comment by qligier** [comment]
Hi @Healthedata1! I guess this is one example of the failure: https://github.com/Healthedata1/R5ElementExtensionTest/actions/runs/5885731783/job/15962594524?

I have created a branch with some modifications:

- The JDK is now Zulu, which is said to be a bit more stable for memory usage.
- There is a step that shows the default configuration of Java.
- There is an input `java-flags` that you can use to configure Java.

You can use it with:
```
- uses: qligier/fhir-ig-action@feature_memory_debug
  with:
    java-flags: "-Xmx4G"
```

 `-Xmx4G` seems the first solution to try, there are other flags that may be useful too.
Otherwise, the IG Publisher should be optimized (e.g. it seems to use a StringBuilder to create file contents, a FileOutputStream would greatly lower the memory usage of this task).

Can you try if you can build your IG with these modifications?

**Comment by Healthedata1** [comment]
Yes I will,  I created a fork with the heap size in the bash command and it
worked so I anticipate this will fix it.

Thanks,

Eric M Haas, DVM, MS
Health eData Inc
211 S Jefferson St, Napa, CA 94559
707.227.2608
***@***.***

**Comment by Healthedata1** [comment]
That worked and thank you!

FYI this is my workflow file:

~~~yaml
# This is a basic workflow to build a FHIR-IG and publish the output (docs folder) 
# on Git Hub Pages everytime it is pushed to the main branch

name: Build and Deploy

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "main" branch
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

permissions:
  contents: write
jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout 🛎️
        uses: actions/checkout@v3
      - name: FHIR IG Action   # action from the GitHub Marketplace
        # uses: Healthedata1/fhir-ig-action@v0.3.0-alpha
        uses: qligier/fhir-ig-action@feature_memory_debug
        with:
          java-flags: "-Xmx4G"
          sushi: "3.2.0"
      - name: Deploy to GitHub Pages  # action from the GitHub Marketplace
        uses: JamesIves/github-pages-deploy-action@v4.4.3
        with:
          folder: docs # deploy from the docs from the prior step
~~~

**Comment by qligier** [comment]
Thank you for testing! I'll then apply the "zulu" and "-Xmx4G" modifications to the action.

**Comment by qligier** [comment]
I've released [v0.3.0](https://github.com/qligier/fhir-ig-action/releases/tag/v0.3.0) with these changes.

**Comment by Healthedata1** [comment]
Ok, thanks.
Eric M Haas, DVM, MS
Health eData Inc
211 S Jefferson St, Napa, CA 94559
707.227.2608
***@***.***
